packages/firestore/test/unit/specs/garbage_collection_spec.test.ts

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { doc, filter, query } from '../../util/helpers';
+
+import { describeSpec, specTest } from './describe_spec';
+import { spec } from './spec_builder';
+
+describeSpec('Garbage Collection:', [], () => {
+  specTest(
+    'Contents of query are cleared when listen is removed.',
+    ['eager-gc'],
+    'Explicitly tests eager GC behavior',
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { key: 'a' });
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA)
+          .expectEvents(query1, { added: [docA] })
+          .userUnlistens(query1)
+          // should get no events.
+          .userListens(query1)
+      );
+    }
+  );
+
+  specTest('Contents of query are kept after listen is removed.', [], () => {
+    const query1 = query('collection');
+    const docA = doc('collection/a', 1000, { key: 'a' });
+    return spec()
+      .ensureManualLruGC()
+      .userListens(query1)
+      .watchAcksFull(query1, 1000, docA)
+      .expectEvents(query1, { added: [docA] })
+      .userUnlistens(query1)
+      .userListens(query1)
+      .expectListen(query1, { resumeToken: 'resume-token-1000' })
+      .expectEvents(query1, { added: [docA], fromCache: true });
+  });
+
+  specTest(
+    'Contents of query are kept after listen is removed, and GC threshold is not reached',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { key: 'a' });
+      return spec()
+        .ensureManualLruGC()
+        .userListens(query1)
+        .watchAcksFull(query1, 1000, docA)
+        .expectEvents(query1, { added: [docA] })
+        .userUnlistens(query1)
+        .triggerLruGC(1000_000_000)
+        .userListens(query1)
+        .expectListen(query1, { resumeToken: 'resume-token-1000' })
+        .expectEvents(query1, { added: [docA], fromCache: true });
+    }
+  );
+
+  specTest(
+    'Contents of query are kept after listen is removed, and GC threshold is reached',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { key: 'a' });
+      return (
+        spec()
+          .ensureManualLruGC()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA)
+          .expectEvents(query1, { added: [docA] })
+          .userUnlistens(query1)
+          .triggerLruGC(1)
+          .removeExpectedTargetMapping(query1)
+          // should get no events.
+          .userListens(query1)
```

Comment:
Is the test name and the test implementation aligned on this one? Maybe I'm not following what it's trying to assert

Reply:
Nope, fixed.